test/arrays/types-resized-array/testTuple.future

```diff
@@ -0,0 +1,2 @@
+Array of tuple emits default initialization errors instead of halting
```

Comment:
@mppf Unless you can think of a quick way to handle this, for now I'm just planning on migrating this test to a future because the expected behavior (halt at first domain resize) doesn't work for arrays of tuples of non-nilable classes.

Reply:
One possible solution is to allow `pragma "unsafe"` to skip the checks I added to tuple default init and just blindly insert the call to `_defaultOf`, but that just returns us to the previous error. So I view this new `.bad` as more precise rather than different.

Reply:
Yes I think that investigating the issue with such arrays of tuples can be a separate effort (and in particular it might need adjustments in different places). The change to the future here seems OK to me.